
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">

    <title>Aldy | Algorithms for Cancer</title>
    <meta name="description" content="Cenk Sahinalp Lab">
    <meta name="author" content="Cenk Sahinalp">

    <!-- Enable responsive viewport -->
    <meta name="viewport" content="width=device-width, initial-scale=1.0"/>

    <!-- Bootstrap styles -->
    <link href="https://algo-cancer.github.io/assets/themes/lab/bootstrap/css/bootstrap.min.css" rel="stylesheet"/>
    <!-- Optional theme -->
    <link href="https://algo-cancer.github.io/assets/themes/lab/bootstrap/css/bootstrap-theme.min.css" rel="stylesheet"/>
    <!-- Sticky Footer -->
    <link href="https://algo-cancer.github.io/assets/themes/lab/bootstrap/css/bs-sticky-footer.css" rel="stylesheet"/>
    
    <!-- Custom styles -->
    <link href="https://algo-cancer.github.io/assets/themes/lab/css/style.css?body=1" rel="stylesheet" type="text/css" media="all"/>

    <!-- icon -->
    <link rel="icon" type="image/png" href="https://algo-cancer.github.io/assets/themes/lab/images/logo/favicon.png"/>

    <!-- Custom fonts and icons via Font Awesome, http://fortawesome.github.io/Font-Awesome/ -->
    <link rel="stylesheet" href="http://maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css"/>

    <!-- Fonts via Google -->
    <link href='http://fonts.googleapis.com/css?family=Lato:300italic,700italic,300,700' rel='stylesheet' type='text/css'/>

    <!-- Math via MathJax -->
    <script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>


    <!-- HTML5 Shim and Respond.js IE8 support of HTML5 elements and media queries -->
    <!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
    <!--[if lt IE 9]>
      <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
      <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
    <![endif]-->

    <!-- Fav and touch icons -->
    <!-- Update these with your own images
      <link rel="shortcut icon" href="images/favicon.ico">
      <link rel="apple-touch-icon" href="images/apple-touch-icon.png">
      <link rel="apple-touch-icon" sizes="72x72" href="images/apple-touch-icon-72x72.png">
      <link rel="apple-touch-icon" sizes="114x114" href="images/apple-touch-icon-114x114.png">
    -->

    <!-- atom & rss feed
    <link href="/atom.xml" type="application/atom+xml" rel="alternate" title="Sitewide ATOM Feed">
    <link href="/rss.xml" type="application/rss+xml" rel="alternate" title="Sitewide RSS Feed">
    -->

  </head>

  <body>

  <!-- Static top navbar -->

  <nav class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class='container'>
      <!-- Brand and toggle get grouped for better mobile display -->
      <div class="navbar-header">
        <button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#jb-navbar-collapse">
          <span class="sr-only">Toggle navigation</span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
        <a href="/">
        <img class="pull-left logo" src="https://algo-cancer.github.io/images/logo.png">
        </a>
      </div>

      <!-- Collect the nav links, forms, and other content for toggling -->
      <div class="collapse navbar-collapse" id="jb-navbar-collapse">
        <ul class="nav navbar-nav navbar-left hidden-sm">
          <!-- Don't put site title on home page -- there's already a big title there. -->
          
          
          
          <a class="navbar-brand" href="https://algo-cancer.github.io/">Algorithms for Cancer</a>
          
        </ul>
        <ul class="nav navbar-nav navbar-right pull-right">
          
          
          


  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
  
    
  
    
  
    
  
    
      
    
  
    
      
    
  
    
      
      	
      	
      
    
  
    
      
      	
      	
      
    
  
    
      
      	
      	<li><a href="https://algo-cancer.github.io/team/">Team</a></li>
      	
      
    
  
    
      
      	
      	<li><a href="https://algo-cancer.github.io/papers/">Publications</a></li>
      	
      
    
  
    
      
      	
      	<li><a href="https://algo-cancer.github.io/projects/">Software</a></li>
      	
      
    
  



          <!--<li class="dropdown nav navbar-nav">-->
            <!--<a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">-->
              <!--More <span class="caret"></span>-->
            <!--</a>-->

            <!--<ul class="dropdown-menu nav navbar-nav" role="menu">-->
              <!---->
              <!---->
              <!--


  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
  
    
  
    
  
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  


-->
            <!--</ul>-->
          <!--</li>-->
        </ul>
      </div><!-- /.navbar-collapse -->
    </nav>
   </div>

   <!-- Content -->
   <div class="container">
      
<div class="bigspacer"></div>
<div class="bigspacer"></div>

<div class="row">
	<div class="col-md-12">
		<div class="page-header">
  			<div class="title media-heading">Aldy</div>
  			
		</div>
	</div>
</div>

<div class="bigspacer"></div>

<div class="row">
	<div class="col-md-3">
		<div class="bigspacer"></div>
		<div class="glyphbox note">
			<div class="pad-left note">
				<div class="smallspacer"></div>
				<i class="fa fa-github-alt fa-fw"></i>
				<a class="off" href="https://github.com/inumanag/aldy">Source Code</a>
			</div>
			<div class="bigspacer"></div>
		</div>
	</div>
	<div class="col-md-8">
    	<div class="content">
      	<h1 id="aldy"><strong>Aldy</strong></h1>

<p>Aldy is a tool for allelic decomposition and exact genotyping of highly polymorphic and structurally variant genes.
More simply, it is a tool which can detect the copy number of a target gene, and the structure and genotype of each gene copy present in the sample.</p>

<p>Aldy has been published in <a href="https://www.nature.com/articles/s41467-018-03273-1">Nature Communications</a> (<a href="http://doi.org/10.1038/s41467-018-03273-1">doi:10.1038/s41467-018-03273-1</a>). Preprint <a href="preprint.pdf">is available here</a>. Full experimental pipeline <a href="https://github.com/inumanag/aldy-paper-resources">is available here</a>.</p>

<h2 id="installation">Installation</h2>

<p>Aldy is written in Python, and supports both Python 2.7 and 3. It is intended to be run on POSIX-based systems (only Linux and macOS have been tested).</p>

<p>The easiest way to install Aldy is to use <code class="language-plaintext highlighter-rouge">pip</code>:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>pip install git+https://github.com/inumanag/aldy.git
</code></pre></div></div>

<p>Add <code class="language-plaintext highlighter-rouge">--user</code> to install it locally if you cannot write to the system-wide Python directory.</p>

<h3 id="prerequisite-ilp-solver">Prerequisite: ILP solver</h3>

<p>Although Aldy can be installed without any ILP solver, it cannot produce any meaningful results without supported ILP solver. Unfortunately, as of now, none of the currently supported solvers can be installed automatically.</p>

<p>Solvers which are currently supported include:</p>

<ul>
  <li><a href="http://www.gurobi.com">Gurobi</a> (<strong>highly recommended</strong>):
 a commercial solver which is free for academic purposes.
 After installing it, don’t forget to install <code class="language-plaintext highlighter-rouge">gurobipy</code> package by going to Gurobi’s installation directory (e.g. <code class="language-plaintext highlighter-rouge">/opt/gurobi/linux64</code> on Linux or <code class="language-plaintext highlighter-rouge">/Library/gurobi751/mac64/</code> on macOS) and issuing <code class="language-plaintext highlighter-rouge">python setup.py install</code>.</li>
  <li><a href="http://scip.zib.de/">SCIP</a>:
   another solver that is also free for academic purposes, and that is bit easier to obtain than Gurobi (no registration required). However, please note that it is slower than Gurobi. Please bear in mind that the results from the paper were generated by Gurobi, although SCIP should provide the same results. Once you install SCIP, please install <a href="https://github.com/SCIP-Interfaces/PySCIPOpt">PySCIPPpt</a> module for the Python SCIP bindings.</li>
</ul>

<h3 id="sanity-check">Sanity check</h3>

<p>After installing Aldy and the supported ILP solver, please make sure to test the installation by issuing the following command (this should take around a minute):</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>aldy --test
</code></pre></div></div>

<p>In case everything is set up properly, you should see something like this:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>Aldy Sanity-Check Test
Expected result is: *1/*4+*4

*** Aldy v1.2 (Python 3.6) ***
(c) 2017 SFU, MIT &amp; IUB. All rights reserved.
Arguments:
  Gene:      CYP2D6
  Profile:   illumina
...[redacted]...
Result:
  *1/*4+*4                       (1, 4N, 4AW)
</code></pre></div></div>

<h2 id="running">Running</h2>

<p>Aldy needs an aligned SAM, BAM, CRAM or DeeZ file for the analysis. Below we will use BAM as an example.</p>

<p><strong>It is assumed that reads are mapped to hg19 or GRCh37. hg38 is not yet supported.</strong></p>

<p>An index is needed for BAM files. Get one by running:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>samtools index file.bam
</code></pre></div></div>

<p>Aldy is invoked as:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>aldy -p [profile] -g [gene] file.bam
</code></pre></div></div>

<p>The <code class="language-plaintext highlighter-rouge">[gene]</code> parameter indicates the name of the gene to be genotyped. Currently, Aldy supports <em>CYP2D6</em>, <em>CYP2A6</em>, <em>CYP2C19</em>, <em>CYP2C8</em>, <em>CYP2C9</em>, <em>CYP3A4</em>, <em>CYP3A5</em>, <em>CYP4F2</em>, <em>TPMT</em> and <em>DPYD</em>.</p>

<h3 id="sequencing-profile-selection">Sequencing profile selection</h3>

<p>The <code class="language-plaintext highlighter-rouge">[profile]</code> argument refers to the sequencing profile. The following profiles are available:</p>

<ul>
  <li>
    <p><code class="language-plaintext highlighter-rouge">illumina</code> for Illumina WGS (or any uniform-coverage technology) samples.</p>

    <p><strong>It is highly recommended to use samples with at least 40x coverage. Also, anything lower than 20x will result in tears and agony.</strong></p>
  </li>
  <li><code class="language-plaintext highlighter-rouge">pgrnseq-v1</code> for PGRNseq v.1 capture protocol data</li>
  <li><code class="language-plaintext highlighter-rouge">pgrnseq-v2</code> for PGRNseq v.2 capture protocol data</li>
</ul>

<p>In case you are using different technology (i.e. some home-brewed capture kit), you can use it provided that the following requirements are met:</p>

<ul>
  <li>all samples have the similar coverage distribution (i.e. two sequenced samples with the same copy number configuration MUST have similar coverage profiles; please consult us if you are not sure about this)</li>
  <li>your panel includes a copy-number neutral region (currently, Aldy uses CYP2D8 as a copy-number neutral region)</li>
</ul>

<p>Having said that, you can generate the profile for your panel/technology by running:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c"># Get the profile</span>
aldy <span class="nt">--generate-profile</span> file.bam <span class="o">&gt;</span> my-cool-tech.profile
<span class="c"># Run Aldy</span>
aldy <span class="nt">-p</span> my-cool-tech.profile <span class="nt">-g</span> <span class="o">[</span>gene] file.bam
</code></pre></div></div>

<h3 id="output">Output</h3>

<p>Aldy will generate the following files: <code class="language-plaintext highlighter-rouge">file-[gene].aldy</code> (default location can be changed via <code class="language-plaintext highlighter-rouge">-o</code> parameter), and <code class="language-plaintext highlighter-rouge">file-[gene].aldylog</code> (default location can be changed via <code class="language-plaintext highlighter-rouge">-l</code> parameter).  The summary of results are shown
at the end of the output:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>$ aldy -p pgrnseq-v2 -g cyp2d6 NA19788_x.bam
*** Aldy v1.0 ***
(c) 2017 SFU, MIT &amp; IUB. All rights reserved.
Arguments:
  Gene:      CYP2D6
  Profile:   pgrnseq-v2
  Threshold: 50%
  Input:     NA19788_x.bam
  Output:    NA19788_x.CYP2D6.aldy
  Log:       NA19788_x.CYP2D6.aldylog
Result:
  *2/*78+*2                      (2MW, 2MW, 78/2|2M)
</code></pre></div></div>

<p>In this example, <em>CYP2D6</em> genotype is *2/*78+*2 as expressed in terms of major star-alleles. Minor star-alleles are given in the parenthesis (in this case, two copies of *2MW, and one copy of *78 fusion on the *2M background).</p>

<p>Explicit decomposition is given in the <code class="language-plaintext highlighter-rouge">file-[gene].aldy</code> (in the example above, it is <code class="language-plaintext highlighter-rouge">NA19788_x.CYP2D6.aldy</code>). 
Example of such file is:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code># Aldy v1.0
# Gene: CYP2D6
# Number of solutions: 1

# Solution 0
# Predicted diplotype: *2/*78+*2
# Composition: 2MW,2MW,78/2|2M
Copy   Allele   Location   Type     Coverage  Effect      dbSNP       Code        Status
0      78/2     42522311   SNP.CT   1760      NEUTRAL     rs12169962  4481:G&gt;A    NORMAL
0      78/2     42522612   SNP.CG   1287      DISRUPTING  rs1135840   4180:G&gt;C    NORMAL
...[redacted]...
1      2MW      42522311   SNP.CT   1760      NEUTRAL     rs12169962  4481:G&gt;A    NORMAL
1      2MW      42527541   DEL.TC   0         NEUTRAL     rs536645539 -750:delGA  MISSING
...[redacted]...
</code></pre></div></div>

<p>Each solution is indicated with the <strong>“Solution”</strong> line. The first column (copy) shows the ordinary number of the allelic copy (e.g. 0, 1 and 2 for 2MW, 2MW and 78/2M, respectively). The following columns indicate:</p>

<ul>
  <li>star-allele,</li>
  <li>mutation loci,</li>
  <li>mutation type (SNP or indel),</li>
  <li>mutation coverage,</li>
  <li>mutation functionality:
    <ul>
      <li><code class="language-plaintext highlighter-rouge">DISRUPTING</code> for gene-disrupting</li>
      <li><code class="language-plaintext highlighter-rouge">NEUTRAL</code> for neutral mutation,</li>
    </ul>
  </li>
  <li>dbSNP ID (if available),</li>
  <li>traditional Karolinska-style mutation code from CYP allele database, and</li>
  <li>mutation status, which indicates the status of the mutation in the decomposition:
    <ul>
      <li><code class="language-plaintext highlighter-rouge">NORMAL</code>: mutation is associated with the star-allele in the database, and is found in the sample</li>
      <li><code class="language-plaintext highlighter-rouge">NOVEL</code>: gene-disrupting mutation is <strong>NOT</strong> associated with the star-allele in the database, 
but is found in the sample (this indicates that Aldy found a novel major star-allele)</li>
      <li><code class="language-plaintext highlighter-rouge">EXTRA</code>: neutral mutation is <strong>NOT</strong> associated with the star-allele in the database, 
but is found in the sample (this indicates that Aldy found a novel minor star-allele)</li>
      <li><code class="language-plaintext highlighter-rouge">MISSING</code>: neutral mutation is associated with the star-allele in the database, 
but is <strong>NOT</strong> found in the sample (this also indicates that Aldy found a novel minor star-allele)</li>
    </ul>
  </li>
</ul>

<h3 id="logging">Logging</h3>

<p>Detailed execution log will be located in <code class="language-plaintext highlighter-rouge">file-[gene].aldylog</code>. It is used mainly for debugging purposes.</p>

<h2 id="sample-datasets">Sample datasets</h2>

<p>Sample datasets are also available for download. They include:</p>

<ul>
  <li><a href="http://cb.csail.mit.edu/cb/aldy/data/HG00463.bam">HG00463</a> (PGRNseq v.2), containing <em>CYP2D6</em> configuration with multiple copies</li>
  <li><a href="http://cb.csail.mit.edu/cb/aldy/data/NA19790.bam">NA19790</a> (PGRNseq v.2), containing a fusion between <em>CYP2D6</em> and <em>CYP2D7</em> deletion (*78 allele)</li>
  <li><a href="http://cb.csail.mit.edu/cb/aldy/data/NA24027.bam">NA24027</a> (PGRNseq v.1), containing novel <em>DPYD</em> allele and multiple copies of <em>CYP2D6</em></li>
  <li><a href="http://cb.csail.mit.edu/cb/aldy/data/NA10856.bam">NA10856</a> (PGRNseq v.1), containing <em>CYP2D6</em> deletion (*5 allele)</li>
  <li><a href="http://cb.csail.mit.edu/cb/aldy/data/NA10860.bam">NA10860</a> (Illumina WGS), containing 3 copies of <em>CYP2D6</em>. This sample contains only <em>CYP2D6</em> region.</li>
</ul>

<p>Expected results are:</p>

<table>
  <thead>
    <tr>
      <th>Gene (<code class="language-plaintext highlighter-rouge">-g</code>)</th>
      <th>HG00463</th>
      <th>NA19790</th>
      <th>NA24027</th>
      <th>NA10856</th>
      <th>NA10860</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><em>CYP2D6</em></td>
      <td>*36+*10/*36+*10</td>
      <td>*1/*78+*2</td>
      <td>*6/*2+*2</td>
      <td>*1/*5</td>
      <td>*1/*4+*4</td>
    </tr>
    <tr>
      <td><em>CYP2A6</em></td>
      <td>*1/*1</td>
      <td>*1/*1</td>
      <td>*1/*35</td>
      <td>*1/*1</td>
      <td> </td>
    </tr>
    <tr>
      <td><em>CYP2C19</em></td>
      <td>*1/*3</td>
      <td>*1/*1</td>
      <td>*1/*2</td>
      <td>*1/*2</td>
      <td> </td>
    </tr>
    <tr>
      <td><em>CYP2C8</em></td>
      <td>*1/*1</td>
      <td>*1/*3</td>
      <td>*1/*3</td>
      <td>*1/*1</td>
      <td> </td>
    </tr>
    <tr>
      <td><em>CYP2C9</em></td>
      <td>*1/*1</td>
      <td>*1/*2</td>
      <td>*1/*2</td>
      <td>*1/*2</td>
      <td> </td>
    </tr>
    <tr>
      <td><em>CYP3A4</em></td>
      <td>*1/*1</td>
      <td>*1/*1</td>
      <td>*1/*1</td>
      <td>*1/*1</td>
      <td> </td>
    </tr>
    <tr>
      <td><em>CYP3A5</em></td>
      <td>*3/*3</td>
      <td>*3/*3</td>
      <td>*1/*3</td>
      <td>*1/*3</td>
      <td> </td>
    </tr>
    <tr>
      <td><em>CYP4F2</em></td>
      <td>*1/*1</td>
      <td>*3/*4</td>
      <td>*1/*1</td>
      <td>*1/*1</td>
      <td> </td>
    </tr>
    <tr>
      <td><em>TPMT</em></td>
      <td>*1/*1</td>
      <td>*1/*1</td>
      <td>*1/*1</td>
      <td>*1/*1</td>
      <td> </td>
    </tr>
    <tr>
      <td><em>DPYD</em></td>
      <td>*1/*1</td>
      <td>*1/*1</td>
      <td>*4/*5</td>
      <td>*5/*6</td>
      <td> </td>
    </tr>
  </tbody>
</table>

<h2 id="license">License</h2>

<p>© 2017 Simon Fraser University, Indiana University Bloomington &amp; Massachusetts Institute of Technology. All rights reserved.</p>

<p><strong>Aldy is NOT a free software.</strong> Complete legal license is available in <a href="LICENSE.md">LICENSE</a> file.</p>

<p>For non-legal folks, here is TL;DR version:</p>

<ul>
  <li>Aldy can be freely used in academic and non-commercial environments</li>
  <li>Please contact us if you intend to use Aldy for any commercial purpose</li>
</ul>

<h2 id="parameter-documentation">Parameter documentation</h2>

<p><strong>NAME</strong></p>

<p>aldy — perform allelic decomposition and exact genotyping on next-generation sequencing data</p>

<p><strong>SYNOPSIS</strong></p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>aldy -h
aldy --test
aldy --license
aldy --generate-profile
aldy --show-cn --gene
aldy [--threshold THRESHOLD] [--profile PROFILE]
     [--gene GENE] [--verbosity VERBOSITY] [--log LOG]
     [--output OUTPUT] [--solver SOLVER] [--phase] [--cn CN]
     [file]
</code></pre></div></div>

<p><strong>OPTIONS</strong></p>

<p>Positional arguments:</p>
<ul>
  <li><code class="language-plaintext highlighter-rouge">file</code>                <br />
  SAM, BAM, CRAM or DeeZ input file</li>
</ul>

<p>Optional arguments:</p>

<ul>
  <li><code class="language-plaintext highlighter-rouge">-h, --help </code><br />
 Show the help message and exit</li>
  <li><code class="language-plaintext highlighter-rouge">-T, --threshold THRESHOLD</code><br />
 Cut-off rate for variations (percent per copy)<br />
 <em>default:</em> <code class="language-plaintext highlighter-rouge">50</code></li>
  <li><code class="language-plaintext highlighter-rouge">-p, --profile PROFILE</code><br />
 Sequencing profile. Currently, only “illumina”, “pgrnseq-v1” and “pgrnseq-v2” are supported. Please check <code class="language-plaintext highlighter-rouge">--generate-profile</code> for more information how to use your own profile</li>
  <li><code class="language-plaintext highlighter-rouge">-g, --gene GENE</code><br />
 Gene profile<br />
 <em>default:</em> <code class="language-plaintext highlighter-rouge">CYP2D6</code></li>
  <li><code class="language-plaintext highlighter-rouge">-v, --verbosity VERBOSITY</code><br />
 Logging verbosity. Acceptable values are T (trace), D (debug), I (info) and W (warn)<br />
 <em>default:</em> <code class="language-plaintext highlighter-rouge">I</code></li>
  <li><code class="language-plaintext highlighter-rouge">-l, --log LOG</code><br />
 Location of the output log file <br />
 <em>default:</em> <code class="language-plaintext highlighter-rouge">[input].[gene].aldylog</code></li>
  <li><code class="language-plaintext highlighter-rouge">-o, --output OUTPUT</code><br />
 Location of the output file <br />
 <em>default:</em> <code class="language-plaintext highlighter-rouge">[input].[gene].aldy</code></li>
  <li><code class="language-plaintext highlighter-rouge">-s, --solver SOLVER</code><br />
 IP Solver. Currently supported solvers are Gurobi and SCIP  <br />
 <em>default:</em> <code class="language-plaintext highlighter-rouge">any</code></li>
  <li><code class="language-plaintext highlighter-rouge">-P, --phase</code><br />
 Phase reads<br />
 <em>default:</em> <code class="language-plaintext highlighter-rouge">no</code> (slows down the pipeline)</li>
  <li><code class="language-plaintext highlighter-rouge">--license</code><br />
 Print Aldy license</li>
  <li><code class="language-plaintext highlighter-rouge">--test</code><br />
 Sanity-check on NA10860 sample</li>
  <li><code class="language-plaintext highlighter-rouge">--show-cn</code><br />
 Show all copy number configurations supported by a gene (requires <code class="language-plaintext highlighter-rouge">-g</code>)</li>
  <li><code class="language-plaintext highlighter-rouge">-c, --cn CN</code><br />
 Manually set copy number (input: a comma-separated list CN1,CN2,…). For a list of supported configurations, please run <code class="language-plaintext highlighter-rouge">aldy --show-cn</code></li>
  <li><code class="language-plaintext highlighter-rouge">--generate-profile </code> <br />
 Generate the copy-number profile for the custom sequencing panel and print it on the standard output</li>
</ul>

<h2 id="contact--bug-reports">Contact &amp; Bug Reports</h2>

<p><a href="mailto:inumanag.at.mit.dot.edu">Ibrahim Numanagić</a></p>

<p>If you have an urgent question, I suggest using e-mail: GitHub issues are not handled as fast as the email requests.</p>

    	</div> 
	</div>
	<div class="col-md-1"></div>
</div>
<div class="bigspacer"></div>


   </div>


    



    <!-- Latest compiled and minified JavaScript, requires jQuery 1.x (2.x not supported in IE8) -->
    <!-- Placed at the end of the document so the pages load faster -->
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.10.2/jquery.min.js"></script>
    <script src="https://algo-cancer.github.io/assets/themes/lab/bootstrap/js/bootstrap.min.js"></script>
    <script type='text/javascript' src='https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js'></script>
  </body>
</html>

