
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">

    <title>CAMMiQ | Algorithms for Cancer</title>
    <meta name="description" content="Cenk Sahinalp Lab">
    <meta name="author" content="Cenk Sahinalp">

    <!-- Enable responsive viewport -->
    <meta name="viewport" content="width=device-width, initial-scale=1.0"/>

    <!-- Bootstrap styles -->
    <link href="https://algo-cancer.github.io/assets/themes/lab/bootstrap/css/bootstrap.min.css" rel="stylesheet"/>
    <!-- Optional theme -->
    <link href="https://algo-cancer.github.io/assets/themes/lab/bootstrap/css/bootstrap-theme.min.css" rel="stylesheet"/>
    <!-- Sticky Footer -->
    <link href="https://algo-cancer.github.io/assets/themes/lab/bootstrap/css/bs-sticky-footer.css" rel="stylesheet"/>
    
    <!-- Custom styles -->
    <link href="https://algo-cancer.github.io/assets/themes/lab/css/style.css?body=1" rel="stylesheet" type="text/css" media="all"/>

    <!-- icon -->
    <link rel="icon" type="image/png" href="https://algo-cancer.github.io/assets/themes/lab/images/logo/favicon.png"/>

    <!-- Custom fonts and icons via Font Awesome, http://fortawesome.github.io/Font-Awesome/ -->
    <link rel="stylesheet" href="http://maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css"/>

    <!-- Fonts via Google -->
    <link href='http://fonts.googleapis.com/css?family=Lato:300italic,700italic,300,700' rel='stylesheet' type='text/css'/>

    <!-- Math via MathJax -->
    <script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>


    <!-- HTML5 Shim and Respond.js IE8 support of HTML5 elements and media queries -->
    <!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
    <!--[if lt IE 9]>
      <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
      <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
    <![endif]-->

    <!-- Fav and touch icons -->
    <!-- Update these with your own images
      <link rel="shortcut icon" href="images/favicon.ico">
      <link rel="apple-touch-icon" href="images/apple-touch-icon.png">
      <link rel="apple-touch-icon" sizes="72x72" href="images/apple-touch-icon-72x72.png">
      <link rel="apple-touch-icon" sizes="114x114" href="images/apple-touch-icon-114x114.png">
    -->

    <!-- atom & rss feed
    <link href="/atom.xml" type="application/atom+xml" rel="alternate" title="Sitewide ATOM Feed">
    <link href="/rss.xml" type="application/rss+xml" rel="alternate" title="Sitewide RSS Feed">
    -->

  </head>

  <body>

  <!-- Static top navbar -->

  <nav class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class='container'>
      <!-- Brand and toggle get grouped for better mobile display -->
      <div class="navbar-header">
        <button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#jb-navbar-collapse">
          <span class="sr-only">Toggle navigation</span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
        <a href="/">
        <img class="pull-left logo" src="https://algo-cancer.github.io/images/logo.png">
        </a>
      </div>

      <!-- Collect the nav links, forms, and other content for toggling -->
      <div class="collapse navbar-collapse" id="jb-navbar-collapse">
        <ul class="nav navbar-nav navbar-left hidden-sm">
          <!-- Don't put site title on home page -- there's already a big title there. -->
          
          
          
          <a class="navbar-brand" href="https://algo-cancer.github.io/">Algorithms for Cancer</a>
          
        </ul>
        <ul class="nav navbar-nav navbar-right pull-right">
          
          
          


  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
  
    
  
    
  
    
  
    
      
    
  
    
      
    
  
    
      
      	
      	
      
    
  
    
      
      	
      	
      
    
  
    
      
      	
      	<li><a href="https://algo-cancer.github.io/team/">Team</a></li>
      	
      
    
  
    
      
      	
      	<li><a href="https://algo-cancer.github.io/papers/">Publications</a></li>
      	
      
    
  
    
      
      	
      	<li><a href="https://algo-cancer.github.io/projects/">Software</a></li>
      	
      
    
  



          <!--<li class="dropdown nav navbar-nav">-->
            <!--<a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">-->
              <!--More <span class="caret"></span>-->
            <!--</a>-->

            <!--<ul class="dropdown-menu nav navbar-nav" role="menu">-->
              <!---->
              <!---->
              <!--


  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
  
    
  
    
  
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  


-->
            <!--</ul>-->
          <!--</li>-->
        </ul>
      </div><!-- /.navbar-collapse -->
    </nav>
   </div>

   <!-- Content -->
   <div class="container">
      
<div class="bigspacer"></div>
<div class="bigspacer"></div>

<div class="row">
	<div class="col-md-12">
		<div class="page-header">
  			<div class="title media-heading">CAMMiQ</div>
  			
		</div>
	</div>
</div>

<div class="bigspacer"></div>

<div class="row">
	<div class="col-md-3">
		<div class="bigspacer"></div>
		<div class="glyphbox note">
			<div class="pad-left note">
				<div class="smallspacer"></div>
				<i class="fa fa-github-alt fa-fw"></i>
				<a class="off" href="https://github.com/algo-cancer/CAMMiQ">Source Code</a>
			</div>
			<div class="bigspacer"></div>
		</div>
	</div>
	<div class="col-md-8">
    	<div class="content">
      	<h1 id="cammiq">CAMMiQ</h1>

<h3 id="what-is-cammiq">What is CAMMiQ?</h3>

<p>CAMMiQ is a software tool for microbial identification and quantification. Specifically, it builds a compact index to manage a database of microbial genomes (in particular, bacterial/viral genomes in NCBI <a href="https://www.ncbi.nlm.nih.gov/refseq/">RefSeq</a> database) and answers queries of the following form: given a set of high throughput sequencing (HTS) reads obtained from a mixture of genomes in the database, identify the corresponding genomes (in the database) and compute their relative abundance.</p>

<h3 id="how-to-install-cammiq">How to install CAMMiQ?</h3>
<p>Dependencies: Though several scripts (to compile the code and download the database) are written using the Bash shell and python, our core programs to build the index and query the set of reads are written in C++11, and need to be compiled using a relatively recent version (The current build of CAMMiQ was tested with version 5.2.0) of gcc that will support C++11. Multithreading is handled using pthread and <a href="https://en.wikipedia.org/wiki/OpenMP">OpenMP</a>.</p>

<p>In addition, you’ll need the following components to compile the sources:</p>
<ul>
  <li>https://github.com/jlabeit/parallel-divsufsort which constructs suffix arrays in a parallelized and lightweight fashion.</li>
  <li>https://github.com/martinus/robin-hood-hashing which is a faster and more memory efficient alternative of STL unordered_map.</li>
  <li><a href="https://www.ibm.com/products/ilog-cplex-optimization-studio">IBM ILOG CPLEX Optimization Studio</a> - CAMMiQ requires its c++ interface for solving a (mixed) integer linear program (ILP) to figure out the most likely composition of genomes in the mixture based on the distribution of querying reads.</li>
</ul>

<p>To install CAMMiQ, just clone our repository and run the script install_CAMMiQ.sh we have prepared - it will automatically download the related repos above and compile their codes. The only exception is, you’ll have to first download and install the latest version (e.g., 12.9.0) of IBM ILOG CPLEX Optimization Studio by yourself.</p>
<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>git clone https://github.com/algo-cancer/CAMMiQ
./install_CAMMiQ.sh --cplex-dir &lt;CPLEX_DIR&gt;
</code></pre></div></div>
<p>where <code class="language-plaintext highlighter-rouge">&lt;CPLEX_DIR&gt;</code> should be replaced with the directory of your CPLEX_Studio.</p>

<h3 id="how-do-i-use-cammiq">How do I use CAMMiQ?</h3>
<p>To begin using CAMMiQ, you will first need to index the input genomes.</p>

<h4 id="what-is-cammiq-index-composed-of">What is CAMMiQ index composed of?</h4>
<p>CAMMiQ index is composed of three parts - all of them are necessary to query the input reads:</p>
<ul>
  <li>(i) A set of representative (i.e., maximally sparsified) <em>shortest unique substrings</em> on each input genome, organized as a prefix tree (trie) and hashed based on their common prefixes, encoded in binary file <code class="language-plaintext highlighter-rouge">index_u.bin1</code>.</li>
  <li>(ii) A set of representative (i.e., maximally sparsified) <em>shortest doubly unique substrings</em> on each input genome, organized as a prefix tree (trie) and hashed based on their common prefixes, encoded in binary file <code class="language-plaintext highlighter-rouge">index_d.bin2</code>.</li>
  <li>(iii) Other meta-information of the input genomes, including <code class="language-plaintext highlighter-rouge">genome_lengths.out</code>, a text file containing the genome lengths; <code class="language-plaintext highlighter-rouge">unique_lmer_count_u.out</code>, a text file containing the number of unique <em>L</em>-mers on each genome; and <code class="language-plaintext highlighter-rouge">unique_lmer_count_d.out</code>, a text file containing the number of doubly unique <em>L</em>-mers on each genome.</li>
</ul>

<h4 id="how-do-i-construct-the-index">How do I construct the index?</h4>
<p>You’ll need to run <code class="language-plaintext highlighter-rouge">./cammiq --build [options]</code> from command line, where <code class="language-plaintext highlighter-rouge">[options]</code> specifies the following list of (possibly mandatory) parameters.</p>
<ul>
  <li><code class="language-plaintext highlighter-rouge">-f &lt;MAP_FILE&gt;</code></li>
  <li><code class="language-plaintext highlighter-rouge">-d &lt;FASTA_DIR&gt;</code></li>
  <li><code class="language-plaintext highlighter-rouge">-k &lt;int&gt;</code></li>
  <li><code class="language-plaintext highlighter-rouge">-L &lt;int&gt;</code></li>
  <li><code class="language-plaintext highlighter-rouge">-Lmax &lt;int&gt;</code></li>
  <li><code class="language-plaintext highlighter-rouge">-h &lt;int&gt;|&lt;int1 int2&gt;</code></li>
  <li><code class="language-plaintext highlighter-rouge">-i unique|doubly_unique|both</code></li>
  <li><code class="language-plaintext highlighter-rouge">-t &lt;int&gt;</code></li>
</ul>

    	</div> 
	</div>
	<div class="col-md-1"></div>
</div>
<div class="bigspacer"></div>


   </div>


    



    <!-- Latest compiled and minified JavaScript, requires jQuery 1.x (2.x not supported in IE8) -->
    <!-- Placed at the end of the document so the pages load faster -->
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.10.2/jquery.min.js"></script>
    <script src="https://algo-cancer.github.io/assets/themes/lab/bootstrap/js/bootstrap.min.js"></script>
    <script type='text/javascript' src='https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js'></script>
  </body>
</html>

