
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">

    <title>CoLoRMap | Algorithms for Cancer</title>
    <meta name="description" content="Cenk Sahinalp Lab">
    <meta name="author" content="Cenk Sahinalp">

    <!-- Enable responsive viewport -->
    <meta name="viewport" content="width=device-width, initial-scale=1.0"/>

    <!-- Bootstrap styles -->
    <link href="https://algo-cancer.github.io/assets/themes/lab/bootstrap/css/bootstrap.min.css" rel="stylesheet"/>
    <!-- Optional theme -->
    <link href="https://algo-cancer.github.io/assets/themes/lab/bootstrap/css/bootstrap-theme.min.css" rel="stylesheet"/>
    <!-- Sticky Footer -->
    <link href="https://algo-cancer.github.io/assets/themes/lab/bootstrap/css/bs-sticky-footer.css" rel="stylesheet"/>
    
    <!-- Custom styles -->
    <link href="https://algo-cancer.github.io/assets/themes/lab/css/style.css?body=1" rel="stylesheet" type="text/css" media="all"/>

    <!-- icon -->
    <link rel="icon" type="image/png" href="https://algo-cancer.github.io/assets/themes/lab/images/logo/favicon.png"/>

    <!-- Custom fonts and icons via Font Awesome, http://fortawesome.github.io/Font-Awesome/ -->
    <link rel="stylesheet" href="http://maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css"/>

    <!-- Fonts via Google -->
    <link href='http://fonts.googleapis.com/css?family=Lato:300italic,700italic,300,700' rel='stylesheet' type='text/css'/>

    <!-- Math via MathJax -->
    <script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>


    <!-- HTML5 Shim and Respond.js IE8 support of HTML5 elements and media queries -->
    <!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
    <!--[if lt IE 9]>
      <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
      <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
    <![endif]-->

    <!-- Fav and touch icons -->
    <!-- Update these with your own images
      <link rel="shortcut icon" href="images/favicon.ico">
      <link rel="apple-touch-icon" href="images/apple-touch-icon.png">
      <link rel="apple-touch-icon" sizes="72x72" href="images/apple-touch-icon-72x72.png">
      <link rel="apple-touch-icon" sizes="114x114" href="images/apple-touch-icon-114x114.png">
    -->

    <!-- atom & rss feed
    <link href="/atom.xml" type="application/atom+xml" rel="alternate" title="Sitewide ATOM Feed">
    <link href="/rss.xml" type="application/rss+xml" rel="alternate" title="Sitewide RSS Feed">
    -->

  </head>

  <body>

  <!-- Static top navbar -->

  <nav class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class='container'>
      <!-- Brand and toggle get grouped for better mobile display -->
      <div class="navbar-header">
        <button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#jb-navbar-collapse">
          <span class="sr-only">Toggle navigation</span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
        <a href="/">
        <img class="pull-left logo" src="https://algo-cancer.github.io/images/logo.png">
        </a>
      </div>

      <!-- Collect the nav links, forms, and other content for toggling -->
      <div class="collapse navbar-collapse" id="jb-navbar-collapse">
        <ul class="nav navbar-nav navbar-left hidden-sm">
          <!-- Don't put site title on home page -- there's already a big title there. -->
          
          
          
          <a class="navbar-brand" href="https://algo-cancer.github.io/">Algorithms for Cancer</a>
          
        </ul>
        <ul class="nav navbar-nav navbar-right pull-right">
          
          
          


  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
  
    
  
    
  
    
  
    
      
    
  
    
      
    
  
    
      
      	
      	
      
    
  
    
      
      	
      	
      
    
  
    
      
      	
      	<li><a href="https://algo-cancer.github.io/team/">Team</a></li>
      	
      
    
  
    
      
      	
      	<li><a href="https://algo-cancer.github.io/papers/">Publications</a></li>
      	
      
    
  
    
      
      	
      	<li><a href="https://algo-cancer.github.io/projects/">Software</a></li>
      	
      
    
  



          <!--<li class="dropdown nav navbar-nav">-->
            <!--<a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">-->
              <!--More <span class="caret"></span>-->
            <!--</a>-->

            <!--<ul class="dropdown-menu nav navbar-nav" role="menu">-->
              <!---->
              <!---->
              <!--


  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
  
    
  
    
  
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  


-->
            <!--</ul>-->
          <!--</li>-->
        </ul>
      </div><!-- /.navbar-collapse -->
    </nav>
   </div>

   <!-- Content -->
   <div class="container">
      
<div class="bigspacer"></div>
<div class="bigspacer"></div>

<div class="row">
	<div class="col-md-12">
		<div class="page-header">
  			<div class="title media-heading">CoLoRMap</div>
  			
		</div>
	</div>
</div>

<div class="bigspacer"></div>

<div class="row">
	<div class="col-md-3">
		<div class="bigspacer"></div>
		<div class="glyphbox note">
			<div class="pad-left note">
				<div class="smallspacer"></div>
				<i class="fa fa-github-alt fa-fw"></i>
				<a class="off" href="https://github.com/sfu-compbio/colormap">Source Code</a>
			</div>
			<div class="bigspacer"></div>
		</div>
	</div>
	<div class="col-md-8">
    	<div class="content">
      	<h1 id="colormap"><strong>CoLoRMap</strong></h1>

<h2 id="installation">Installation</h2>
<p>In order to install CoLoRMap, you should first fetch the source code from CoLoRMap git repository.</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>git clone <span class="nt">--recursive</span> https://github.com/sfu-compbio/colormap.git
</code></pre></div></div>

<p>Please note that command argument <code class="language-plaintext highlighter-rouge">--recursive</code> is necessary for downloading submodules automatically. After obtaining the code, you need to install the dependencies. CoLoRMap uses <em>BWA</em>, <em>SAMtools</em>, and <em>Minia</em>. In order to build these dependencies, change to the source directory <code class="language-plaintext highlighter-rouge">colormap</code> and use <code class="language-plaintext highlighter-rouge">make deps</code> command.</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nb">cd </span>colormap
make deps
</code></pre></div></div>

<p>At last, you can compile CoLoRMap binaries simply by running <code class="language-plaintext highlighter-rouge">make</code> command.</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>make
</code></pre></div></div>

<p>CoLoRMap corrects long reads in two different steps: (i) using a shortest path (SP) algorithm. (ii) using an One-End Anchor (OEA) based algorithm.</p>

<h2 id="preparing-the-short-read-data">Preparing the short read data</h2>
<p>SP algorithm does not need paired-end information, but OEA algorithm actually uses paired-end information. In both cases, the program expects to be fed with a single short read file. In case of OEA algorithm, program expects paired-end short reads in interleaved/interlaced format.</p>

<p>Usually, paired-end short reads are stored in two different files. A single interleaved/interlaced read file can be obtained using <code class="language-plaintext highlighter-rouge">fastUtils</code> program which can be found in <code class="language-plaintext highlighter-rouge">bin</code> directory after building the program:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nb">cd </span>testData
../bin/fastUtils shuffle <span class="nt">-1</span> ill_1.fastq <span class="nt">-2</span> ill_2.fastq <span class="nt">-o</span> ill.fastq
</code></pre></div></div>

<h2 id="correcting-long-reads">Correcting long reads</h2>
<p>To correct long reads, you can use <strong>runCorr.sh</strong> script:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>../runCorr.sh pac.fasta ill.fastq testCorr pre 4
</code></pre></div></div>
<p>This runs shortest path correction algorithm for long reads stored in <code class="language-plaintext highlighter-rouge">pac.fasta</code> by short reads stored in <code class="language-plaintext highlighter-rouge">ill.fastq</code> using 4 threads. When this is done, the corrected long reads are stored in <code class="language-plaintext highlighter-rouge">testCorr/pre_sp.fasta</code> file.</p>

<h2 id="improving-the-correction-using-one-end-anchors-oeas">Improving the correction using One-End Anchors (OEAs)</h2>
<p>The script <strong>runOEA.sh</strong> can be used to further improve the quality of corrected long reads by using One-End Anchors (OEAs) to extend the borders of the corrected regions.</p>
<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>../runOEA.sh testCorr/pre_sp.fasta ill.fastq testOEA pre 4
</code></pre></div></div>
<p>This runs OEA algorithm for pre-corrected long reads stored in <code class="language-plaintext highlighter-rouge">testCorr/pre_sp.fasta</code> by paired-end short reads stored in interleaved/interlaced format in <code class="language-plaintext highlighter-rouge">ill.fastq</code> using 4 threads. When this is done, the corrected long reads are stored in <code class="language-plaintext highlighter-rouge">testOEA/pre_oea.fasta</code> file.</p>

<h2 id="publication">Publication</h2>
<p><em>Haghshenas E., Hach F., Sahinalp S.C. and Chauve C.</em>, “CoLoRMap: Correcting Long Reads by Mapping short reads” <a href="http://bioinformatics.oxfordjournals.org/content/32/17/i545.short">Bioinformatics</a> (2016) 32 (17): i545-i551
DOI: <a href="http://dx.doi.org/10.1093/bioinformatics/btw463">10.1093/bioinformatics/btw463</a></p>

<h2 id="contact">Contact</h2>
<p>Please report problems and bugs on <a href="https://github.com/sfu-compbio/colormap/issues">issues page</a>. Otherwise, contact <code class="language-plaintext highlighter-rouge">ehaghshe[at]sfu[dot]ca</code></p>

    	</div> 
	</div>
	<div class="col-md-1"></div>
</div>
<div class="bigspacer"></div>


   </div>


    



    <!-- Latest compiled and minified JavaScript, requires jQuery 1.x (2.x not supported in IE8) -->
    <!-- Placed at the end of the document so the pages load faster -->
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.10.2/jquery.min.js"></script>
    <script src="https://algo-cancer.github.io/assets/themes/lab/bootstrap/js/bootstrap.min.js"></script>
    <script type='text/javascript' src='https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js'></script>
  </body>
</html>

