
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">

    <title>CTPsingle | Algorithms for Cancer</title>
    <meta name="description" content="Cenk Sahinalp Lab">
    <meta name="author" content="Cenk Sahinalp">

    <!-- Enable responsive viewport -->
    <meta name="viewport" content="width=device-width, initial-scale=1.0"/>

    <!-- Bootstrap styles -->
    <link href="https://algo-cancer.github.io/assets/themes/lab/bootstrap/css/bootstrap.min.css" rel="stylesheet"/>
    <!-- Optional theme -->
    <link href="https://algo-cancer.github.io/assets/themes/lab/bootstrap/css/bootstrap-theme.min.css" rel="stylesheet"/>
    <!-- Sticky Footer -->
    <link href="https://algo-cancer.github.io/assets/themes/lab/bootstrap/css/bs-sticky-footer.css" rel="stylesheet"/>
    
    <!-- Custom styles -->
    <link href="https://algo-cancer.github.io/assets/themes/lab/css/style.css?body=1" rel="stylesheet" type="text/css" media="all"/>

    <!-- icon -->
    <link rel="icon" type="image/png" href="https://algo-cancer.github.io/assets/themes/lab/images/logo/favicon.png"/>

    <!-- Custom fonts and icons via Font Awesome, http://fortawesome.github.io/Font-Awesome/ -->
    <link rel="stylesheet" href="http://maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css"/>

    <!-- Fonts via Google -->
    <link href='http://fonts.googleapis.com/css?family=Lato:300italic,700italic,300,700' rel='stylesheet' type='text/css'/>

    <!-- Math via MathJax -->
    <script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>


    <!-- HTML5 Shim and Respond.js IE8 support of HTML5 elements and media queries -->
    <!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
    <!--[if lt IE 9]>
      <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
      <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
    <![endif]-->

    <!-- Fav and touch icons -->
    <!-- Update these with your own images
      <link rel="shortcut icon" href="images/favicon.ico">
      <link rel="apple-touch-icon" href="images/apple-touch-icon.png">
      <link rel="apple-touch-icon" sizes="72x72" href="images/apple-touch-icon-72x72.png">
      <link rel="apple-touch-icon" sizes="114x114" href="images/apple-touch-icon-114x114.png">
    -->

    <!-- atom & rss feed
    <link href="/atom.xml" type="application/atom+xml" rel="alternate" title="Sitewide ATOM Feed">
    <link href="/rss.xml" type="application/rss+xml" rel="alternate" title="Sitewide RSS Feed">
    -->

  </head>

  <body>

  <!-- Static top navbar -->

  <nav class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class='container'>
      <!-- Brand and toggle get grouped for better mobile display -->
      <div class="navbar-header">
        <button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#jb-navbar-collapse">
          <span class="sr-only">Toggle navigation</span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
        <a href="/">
        <img class="pull-left logo" src="https://algo-cancer.github.io/images/logo.png">
        </a>
      </div>

      <!-- Collect the nav links, forms, and other content for toggling -->
      <div class="collapse navbar-collapse" id="jb-navbar-collapse">
        <ul class="nav navbar-nav navbar-left hidden-sm">
          <!-- Don't put site title on home page -- there's already a big title there. -->
          
          
          
          <a class="navbar-brand" href="https://algo-cancer.github.io/">Algorithms for Cancer</a>
          
        </ul>
        <ul class="nav navbar-nav navbar-right pull-right">
          
          
          


  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
  
    
  
    
  
    
  
    
      
    
  
    
      
    
  
    
      
      	
      	<li><a href="https://algo-cancer.github.io/research">Research</a></li>
      	
      
    
  
    
      
      	
      	<li><a href="https://algo-cancer.github.io/blog/">Blog</a></li>
      	
      
    
  
    
      
      	
      	<li><a href="https://algo-cancer.github.io/team/">Team</a></li>
      	
      
    
  
    
      
      	
      	<li><a href="https://algo-cancer.github.io/papers/">Publications</a></li>
      	
      
    
  
    
      
      	
      	<li><a href="https://algo-cancer.github.io/projects/">Software</a></li>
      	
      
    
  



          <!--<li class="dropdown nav navbar-nav">-->
            <!--<a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">-->
              <!--More <span class="caret"></span>-->
            <!--</a>-->

            <!--<ul class="dropdown-menu nav navbar-nav" role="menu">-->
              <!---->
              <!---->
              <!--


  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
  
    
  
    
  
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  


-->
            <!--</ul>-->
          <!--</li>-->
        </ul>
      </div><!-- /.navbar-collapse -->
    </nav>
   </div>

   <!-- Content -->
   <div class="container">
      
<div class="bigspacer"></div>
<div class="bigspacer"></div>

<div class="row">
	<div class="col-md-12">
		<div class="page-header">
  			<div class="title media-heading">CTPsingle</div>
  			
		</div>
	</div>
</div>

<div class="bigspacer"></div>

<div class="row">
	<div class="col-md-3">
		<div class="bigspacer"></div>
		<div class="glyphbox note">
			<div class="pad-left note">
				<div class="smallspacer"></div>
				<i class="fa fa-github-alt fa-fw"></i>
				<a class="off" href="https://github.com/nlgndnmz/CTPsingle">Source Code</a>
			</div>
			<div class="bigspacer"></div>
		</div>
	</div>
	<div class="col-md-8">
    	<div class="content">
      	<h1 id="ctpsingle---clonality-inference-from-low-coverage-single-sample-tumors">CTPsingle - clonality inference from low coverage single-sample tumors</h1>

<h6 id="author-nilgun-donmez">author: Nilgun Donmez</h6>

<p>CTPsingle is a tool that aims to infer the subclonal decomposition using low-coverage sequencing data from a single tumor sample.</p>

<h4 id="installation">INSTALLATION</h4>

<p>The core functionality of CTPsingle is implemented in R. If your system does not have R, you can install it for free-of-charge from https://www.r-project.org/. In addition, CTPsingle requires the following R packages:</p>

<ul>
  <li>ggplot2</li>
  <li>RColorBrewer</li>
  <li>DPpackage</li>
  <li>lpSolve</li>
</ul>

<p>The best way to install these packages is to call the install.packages() function in R. Additionally, CTPsingle comes with several utility scripts written in Python. To run these, you will need Python2.7 and the numpy library.</p>

<h4 id="running">RUNNING</h4>

<p>To run CTPsingle, you need a single file containing mutant read counts formatted as follows:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>
Chromosome Position Mutant Reference Mcount Rcount Multiplier Gender
8 116260205 C A 13 8 2 Female
5 106280608 T C 12 29 2 Female
9 62266747 G C 19 26 2 Female
13 64241231 G A 15 34 2 Female
9 34778288 C G 13 16 2 Female

</code></pre></div></div>

<p>Above, Chromosome and Position give the location of the mutation. Mutant and Reference columns give the variant and reference alleles at this location. Mcount and Rcount give the number of reads supporting the mutation and reference alleles respectively. Multiplier column contains the total copy count of the region containing the position in the tumor sample. Note that since CTPsingle does not handle amplified regions, this column should at most be 2. The Gender column should be one of Female, Male or unknown. If it is set to unknown, mutations on non-autosomal chromosomes are ignored.</p>

<p>In addition to a file formatted as above, you will need a folder containing tree structure files. This folder is supplied with the CTPsingle package and is called “GammaAdjMatrices”. Typical usage of CTPsingle is as follows:</p>

<p>` ./CTPsingle.R -f <mutation_file> -o <output_prefix> -m <adjacency_matrices> `</adjacency_matrices></output_prefix></mutation_file></p>

<p>where ‘mutation_file’ is a file formatted as described and ‘output_prefix’ is a prefix for CTPsingle’s output. ‘adjacency_matrices’ should give the path of the GammaAdjMatrices folder. For your convenience, we also included a simulation dataset under ‘./data’. To test CTPsingle, simply execute the following command from the installation directory:</p>

<p>` ./CTPsingle.R -f ./data/simulation15.frq -o ./data/simulation15 -m ./GammaAdjMatrices `</p>

<p>If execution is successful, the following files will be written under ./data:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>simulation15.png
simulation15_cls.png
simulation15_frq.png
simulation15_cluster_assignments.txt
simulation15_num_3_tree_1.txt
simulation15_num_3_tree_2.txt
</code></pre></div></div>

<p>The simulation15_cluster_assignments.txt contains the assignment of the mutations to subclones. The ‘mostLikely’ column contains the ID of the subclone, while the ‘averageFrequency’ column gives the cancer cell fraction of that subcone. The figures visualize various aspects of the data. The ‘_num_X_Y.txt’ files give possible tree topologies on X number of nodes (i.e. the number of subclones found). The last 4 columns of these files denote:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>... [ parent-node ] [ child-node ] [ cell fraction of child-node ] [ objective score ]
</code></pre></div></div>

<p>which can be used the draw the tree structure. Note that the parent-node of the root node is always given as 0. A smaller objective score means that the data fits that tree structure better. In addition to these files, CTPsingle reports its progress and other statistics to the standard output. The ground truth information for this dataset is also given at ./data/groundTruth15.txt.</p>

<h4 id="utility-scripts">UTILITY SCRIPTS</h4>

<p>CTPsingle package comes with additional python scripts intended for developers only. simulateDataset.py and evaluateDataset.py are internal scripts used to test the CTPsingle implementation. BeyerHedetmieni.py is used to generate the Gamma-Adjacency files under ./GammaAdjMatrices. You can run each script without arguments to see a brief instruction.</p>

    	</div> 
	</div>
	<div class="col-md-1"></div>
</div>
<div class="bigspacer"></div>


   </div>


    



    <!-- Latest compiled and minified JavaScript, requires jQuery 1.x (2.x not supported in IE8) -->
    <!-- Placed at the end of the document so the pages load faster -->
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.10.2/jquery.min.js"></script>
    <script src="https://algo-cancer.github.io/assets/themes/lab/bootstrap/js/bootstrap.min.js"></script>
    <script type='text/javascript' src='https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js'></script>
  </body>
</html>

