
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">

    <title>DeeZ | Algorithms for Cancer</title>
    <meta name="description" content="Cenk Sahinalp Lab">
    <meta name="author" content="Cenk Sahinalp">

    <!-- Enable responsive viewport -->
    <meta name="viewport" content="width=device-width, initial-scale=1.0"/>

    <!-- Bootstrap styles -->
    <link href="https://algo-cancer.github.io/assets/themes/lab/bootstrap/css/bootstrap.min.css" rel="stylesheet"/>
    <!-- Optional theme -->
    <link href="https://algo-cancer.github.io/assets/themes/lab/bootstrap/css/bootstrap-theme.min.css" rel="stylesheet"/>
    <!-- Sticky Footer -->
    <link href="https://algo-cancer.github.io/assets/themes/lab/bootstrap/css/bs-sticky-footer.css" rel="stylesheet"/>
    
    <!-- Custom styles -->
    <link href="https://algo-cancer.github.io/assets/themes/lab/css/style.css?body=1" rel="stylesheet" type="text/css" media="all"/>

    <!-- icon -->
    <link rel="icon" type="image/png" href="https://algo-cancer.github.io/assets/themes/lab/images/logo/favicon.png"/>

    <!-- Custom fonts and icons via Font Awesome, http://fortawesome.github.io/Font-Awesome/ -->
    <link rel="stylesheet" href="http://maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css"/>

    <!-- Fonts via Google -->
    <link href='http://fonts.googleapis.com/css?family=Lato:300italic,700italic,300,700' rel='stylesheet' type='text/css'/>

    <!-- Math via MathJax -->
    <script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>


    <!-- HTML5 Shim and Respond.js IE8 support of HTML5 elements and media queries -->
    <!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
    <!--[if lt IE 9]>
      <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
      <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
    <![endif]-->

    <!-- Fav and touch icons -->
    <!-- Update these with your own images
      <link rel="shortcut icon" href="images/favicon.ico">
      <link rel="apple-touch-icon" href="images/apple-touch-icon.png">
      <link rel="apple-touch-icon" sizes="72x72" href="images/apple-touch-icon-72x72.png">
      <link rel="apple-touch-icon" sizes="114x114" href="images/apple-touch-icon-114x114.png">
    -->

    <!-- atom & rss feed
    <link href="/atom.xml" type="application/atom+xml" rel="alternate" title="Sitewide ATOM Feed">
    <link href="/rss.xml" type="application/rss+xml" rel="alternate" title="Sitewide RSS Feed">
    -->

  </head>

  <body>

  <!-- Static top navbar -->

  <nav class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class='container'>
      <!-- Brand and toggle get grouped for better mobile display -->
      <div class="navbar-header">
        <button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#jb-navbar-collapse">
          <span class="sr-only">Toggle navigation</span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
        <a href="/">
        <img class="pull-left logo" src="https://algo-cancer.github.io/images/logo.png">
        </a>
      </div>

      <!-- Collect the nav links, forms, and other content for toggling -->
      <div class="collapse navbar-collapse" id="jb-navbar-collapse">
        <ul class="nav navbar-nav navbar-left hidden-sm">
          <!-- Don't put site title on home page -- there's already a big title there. -->
          
          
          
          <a class="navbar-brand" href="https://algo-cancer.github.io/">Algorithms for Cancer</a>
          
        </ul>
        <ul class="nav navbar-nav navbar-right pull-right">
          
          
          


  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
  
    
  
    
  
    
  
    
      
    
  
    
      
    
  
    
      
      	
      	
      
    
  
    
      
      	
      	
      
    
  
    
      
      	
      	<li><a href="https://algo-cancer.github.io/team/">Team</a></li>
      	
      
    
  
    
      
      	
      	<li><a href="https://algo-cancer.github.io/papers/">Publications</a></li>
      	
      
    
  
    
      
      	
      	<li><a href="https://algo-cancer.github.io/projects/">Software</a></li>
      	
      
    
  



          <!--<li class="dropdown nav navbar-nav">-->
            <!--<a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">-->
              <!--More <span class="caret"></span>-->
            <!--</a>-->

            <!--<ul class="dropdown-menu nav navbar-nav" role="menu">-->
              <!---->
              <!---->
              <!--


  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
  
    
  
    
  
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  


-->
            <!--</ul>-->
          <!--</li>-->
        </ul>
      </div><!-- /.navbar-collapse -->
    </nav>
   </div>

   <!-- Content -->
   <div class="container">
      
<div class="bigspacer"></div>
<div class="bigspacer"></div>

<div class="row">
	<div class="col-md-12">
		<div class="page-header">
  			<div class="title media-heading">DeeZ</div>
  			
		</div>
	</div>
</div>

<div class="bigspacer"></div>

<div class="row">
	<div class="col-md-3">
		<div class="bigspacer"></div>
		<div class="glyphbox note">
			<div class="pad-left note">
				<div class="smallspacer"></div>
				<i class="fa fa-github-alt fa-fw"></i>
				<a class="off" href="https://github.com/sfu-compbio/deez">Source Code</a>
			</div>
			<div class="bigspacer"></div>
		</div>
	</div>
	<div class="col-md-8">
    	<div class="content">
      	<h1 id="deez-reference-based-compression-by-local-assembly"><strong>DeeZ</strong>: Reference Based Compression by Local Assembly</h1>

<p><strong>Warning</strong>: Current version is 1.9 beta 2. It will eventually become 2.0.
It is huge improvement over 1.x series, but it is still in developement, and it
might contain some rough edges. If you need absolute stability, please use 1.1 
until 2.0 is released.</p>

<h3 id="so-what-is-deez">So what is DeeZ?</h3>
<p>DeeZ is a tool for compressing SAM/BAM files.</p>

<h3 id="how-do-i-get-deez">How do I get DeeZ?</h3>
<p>Just clone our repository and issue <code class="language-plaintext highlighter-rouge">make</code> command:</p>
<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>git clone https://github.com/sfu-compbio/deez.git
cd deez &amp;&amp; make -j 
</code></pre></div></div>
<blockquote>
  <p><strong>Note</strong>: You will need at least g++ 4.8 with C++11 support to compile the sources, as well as BZ2, CURL, and OpenSSL libraries.</p>
</blockquote>

<h3 id="how-do-i-run-deez">How do I run DeeZ?</h3>
<p>DeeZ is invoked as following:</p>

<ul>
  <li>
    <p><strong>Compression</strong></p>

    <p><code class="language-plaintext highlighter-rouge">deez -r [reference] [input.sam] -o [output]</code></p>

    <p>This will compress <code class="language-plaintext highlighter-rouge">input.sam</code> to <code class="language-plaintext highlighter-rouge">input.sam.dz</code>.</p>
  </li>
  <li>
    <p><strong>Decompression</strong></p>

    <p><code class="language-plaintext highlighter-rouge">deez -r [reference] [input.dz] -o [output] ([region])</code></p>

    <p>This will decompress <code class="language-plaintext highlighter-rouge">input.dz</code> to <code class="language-plaintext highlighter-rouge">input.dz.sam</code>. <code class="language-plaintext highlighter-rouge">[region]</code> is optional.</p>
  </li>
  <li>
    <p><strong>Random Access</strong></p>

    <p>You can also specify the region of interest while decompressing (i.e. randomly access the region). For example, to extract some reads from chr16 to standard output, you should run:
 <code class="language-plaintext highlighter-rouge">deez -r [reference] input.dz  -c chr16:15000000-16000000</code></p>
  </li>
</ul>

<p>DeeZ supports URLs as input files. Supported protocols are file://, ftp://, http(s):// and s3://.
For Amazon S3 access, DeeZ accepts <code class="language-plaintext highlighter-rouge">AWS_ACCESS_KEY_ID</code> and <code class="language-plaintext highlighter-rouge">AWS_SECRET_ACCESS_KEY</code> environment 
variables if you need to access private repositories.</p>

<hr />

<h3 id="parameter-explanation">Parameter explanation</h3>

<ul>
  <li>
    <p><code class="language-plaintext highlighter-rouge">--threads, -t [number]</code></p>

    <p>Set up the number of threads DeeZ may use for compression and decompression.</p>

    <p>Default value: <strong>4</strong></p>
  </li>
  <li>
    <p><code class="language-plaintext highlighter-rouge">--header, -h</code></p>

    <p>Outputs the SAM header.</p>
  </li>
  <li>
    <p><code class="language-plaintext highlighter-rouge">--reference, -r [file|directory]</code></p>

    <p>Specify the FASTA reference file.
  If no reference file is provided, DeeZ will either try to download
  reference specified in SAM @SQ:UR field. If such field is not
  available or if reference specified therein is not accessible, 
  DeeZ will use no reference.</p>

    <p>If no reference is being used, compression rate will not be optimal.</p>

    <blockquote>
      <p><strong>Note</strong>: Chromosome names in the SAM and FASTA files must match. Also, instead of one big FASTA file, DeeZ supports reference lookup in the given directory for chr*.fa files, where chr* is the chromosome ID from the SAM file.</p>
    </blockquote>
  </li>
  <li>
    <p><code class="language-plaintext highlighter-rouge">--force, -!</code></p>

    <p>Force overwrite of exiting files.</p>
  </li>
  <li>
    <p><code class="language-plaintext highlighter-rouge">--verbosity, -v [level]</code></p>

    <p>Set the logging level. 
  Level 0 shows only errors and warnings.
  Level 1 shows many other useful information (e.g. download progress,
      compression progress, per-field data usage etc.)
  Level 2 shows debug information, including detailed CURL output.</p>

    <p>Default valie: <strong>0</strong></p>
  </li>
  <li>
    <p><code class="language-plaintext highlighter-rouge">--stdout, -c</code></p>

    <p>Compress/decompress to the stdout.</p>
  </li>
  <li>
    <p><code class="language-plaintext highlighter-rouge">--output, -o [file]</code></p>

    <p>Compress/decompress to the <code class="language-plaintext highlighter-rouge">file</code>.</p>
  </li>
  <li>
    <p><code class="language-plaintext highlighter-rouge">--lossy, -l</code></p>

    <p>Set lossy parameter for quality lossy encoding (for more information, please check <a href="http://scalce.sourceforge.net/">SCALCE</a>).</p>
  </li>
  <li>
    <p><code class="language-plaintext highlighter-rouge">--quality, -q [mode]</code></p>

    <p>If <code class="language-plaintext highlighter-rouge">mode</code> is <strong>1</strong> or <strong>samcomp</strong>, DeeZ will use <a href="https://sourceforge.net/projects/samcomp/">sam_comp</a> quality model to encode the qualities. Quality random access is not supported on those files. 
  If <code class="language-plaintext highlighter-rouge">mode</code> is <strong>2</strong>, DeeZ will use rANS order-2 model. This will result in slightly better compression, but decompression  will be slower.</p>
  </li>
  <li>
    <p><code class="language-plaintext highlighter-rouge">--withflag, -f [flag]</code></p>

    <p>Decompress only mappings which have <code class="language-plaintext highlighter-rouge">flag</code> bits set.</p>
  </li>
  <li>
    <p><code class="language-plaintext highlighter-rouge">--withoutflag, -F [flag]</code></p>

    <p>Decompress only mappings which do not have <code class="language-plaintext highlighter-rouge">flag</code> bits set.</p>
  </li>
  <li>
    <p><code class="language-plaintext highlighter-rouge">--stats, -S</code></p>

    <p>Display mapping statistics (needs DeeZ file as input).</p>
  </li>
  <li>
    <p><code class="language-plaintext highlighter-rouge">--sort, -s</code></p>

    <p>Sort the input SAM/BAM file by mapping location.</p>
  </li>
  <li>
    <p><code class="language-plaintext highlighter-rouge">--sortmem, -M [size]</code></p>

    <p>Maximum memory used for sorting.</p>

    <p>Default value: <strong>1G</strong></p>
  </li>
  <li>
    <p><code class="language-plaintext highlighter-rouge">--noqual, -Q</code></p>

    <p>Disable quality score output during decompression.</p>
  </li>
</ul>

<hr />

<h3 id="contact--support">Contact &amp; Support</h3>

<p>Feel free to drop any inquiry to <a href="mailto:">inumanag at sfu dot ca</a>.</p>

<h3 id="publication">Publication</h3>

<p>DeeZ was publised in <a href="http://www.nature.com/nmeth/journal/v11/n11/full/nmeth.3133.html">Nature Methods in November 2014</a>.</p>

<h3 id="authors">Authors</h3>

<ul>
  <li>Dr. Faraz Hach</li>
  <li>Ibrahim Numanagić</li>
  <li>Dr. S. Cenk Şahinalp</li>
</ul>

<h3 id="many-thanks-to">Many thanks to…</h3>

<ul>
  <li><strong>James Bonfield</strong>, author of <a href="http://sourceforge.net/projects/staden/files/io_lib/">Scramble</a> and <a href="http://sourceforge.net/projects/samcomp/">sam_comp</a> for many useful ideas and suggestions.</li>
  <li><strong>Fabian “ryg” Giessen</strong> for his <a href="https://github.com/rygorous/ryg_rans">rANS implementation</a></li>
  <li><strong>Eugene D. Shelwien</strong> for his <a href="http://compression.ru/sh/aridemo6.rar">arithmetic coder implementation</a></li>
  <li><strong>Vitaliy Vitsentiy</strong> for his <a href="https://github.com/vit-vit/CTPL">ctpl threading library</a></li>
  <li>… and all people who are using and testing DeeZ</li>
</ul>

<h3 id="licence">Licence</h3>

<p>Copyright (c) 2013, 2014, Simon Fraser University, Indiana University Bloomington. All rights reserved.
Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:</p>

<ul>
  <li>Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.</li>
  <li>Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.</li>
  <li>Neither the name of the Simon Fraser University, Indiana University Bloomington nor the names of its contributors may be used to endorse or promote products derived from this software without specific prior written permission.</li>
</ul>

<p>THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS “AS IS” AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.</p>


    	</div> 
	</div>
	<div class="col-md-1"></div>
</div>
<div class="bigspacer"></div>


   </div>


    



    <!-- Latest compiled and minified JavaScript, requires jQuery 1.x (2.x not supported in IE8) -->
    <!-- Placed at the end of the document so the pages load faster -->
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.10.2/jquery.min.js"></script>
    <script src="https://algo-cancer.github.io/assets/themes/lab/bootstrap/js/bootstrap.min.js"></script>
    <script type='text/javascript' src='https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js'></script>
  </body>
</html>

