
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">

    <title>lordFAST | Algorithms for Cancer</title>
    <meta name="description" content="Cenk Sahinalp Lab">
    <meta name="author" content="Cenk Sahinalp">

    <!-- Enable responsive viewport -->
    <meta name="viewport" content="width=device-width, initial-scale=1.0"/>

    <!-- Bootstrap styles -->
    <link href="https://algo-cancer.github.io/assets/themes/lab/bootstrap/css/bootstrap.min.css" rel="stylesheet"/>
    <!-- Optional theme -->
    <link href="https://algo-cancer.github.io/assets/themes/lab/bootstrap/css/bootstrap-theme.min.css" rel="stylesheet"/>
    <!-- Sticky Footer -->
    <link href="https://algo-cancer.github.io/assets/themes/lab/bootstrap/css/bs-sticky-footer.css" rel="stylesheet"/>
    
    <!-- Custom styles -->
    <link href="https://algo-cancer.github.io/assets/themes/lab/css/style.css?body=1" rel="stylesheet" type="text/css" media="all"/>

    <!-- icon -->
    <link rel="icon" type="image/png" href="https://algo-cancer.github.io/assets/themes/lab/images/logo/favicon.png"/>

    <!-- Custom fonts and icons via Font Awesome, http://fortawesome.github.io/Font-Awesome/ -->
    <link rel="stylesheet" href="http://maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css"/>

    <!-- Fonts via Google -->
    <link href='http://fonts.googleapis.com/css?family=Lato:300italic,700italic,300,700' rel='stylesheet' type='text/css'/>

    <!-- Math via MathJax -->
    <script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>


    <!-- HTML5 Shim and Respond.js IE8 support of HTML5 elements and media queries -->
    <!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
    <!--[if lt IE 9]>
      <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
      <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
    <![endif]-->

    <!-- Fav and touch icons -->
    <!-- Update these with your own images
      <link rel="shortcut icon" href="images/favicon.ico">
      <link rel="apple-touch-icon" href="images/apple-touch-icon.png">
      <link rel="apple-touch-icon" sizes="72x72" href="images/apple-touch-icon-72x72.png">
      <link rel="apple-touch-icon" sizes="114x114" href="images/apple-touch-icon-114x114.png">
    -->

    <!-- atom & rss feed
    <link href="/atom.xml" type="application/atom+xml" rel="alternate" title="Sitewide ATOM Feed">
    <link href="/rss.xml" type="application/rss+xml" rel="alternate" title="Sitewide RSS Feed">
    -->

  </head>

  <body>

  <!-- Static top navbar -->

  <nav class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class='container'>
      <!-- Brand and toggle get grouped for better mobile display -->
      <div class="navbar-header">
        <button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#jb-navbar-collapse">
          <span class="sr-only">Toggle navigation</span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
        <a href="/">
        <img class="pull-left logo" src="https://algo-cancer.github.io/images/logo.png">
        </a>
      </div>

      <!-- Collect the nav links, forms, and other content for toggling -->
      <div class="collapse navbar-collapse" id="jb-navbar-collapse">
        <ul class="nav navbar-nav navbar-left hidden-sm">
          <!-- Don't put site title on home page -- there's already a big title there. -->
          
          
          
          <a class="navbar-brand" href="https://algo-cancer.github.io/">Algorithms for Cancer</a>
          
        </ul>
        <ul class="nav navbar-nav navbar-right pull-right">
          
          
          


  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
  
    
  
    
  
    
  
    
      
    
  
    
      
    
  
    
      
      	
      	
      
    
  
    
      
      	
      	
      
    
  
    
      
      	
      	<li><a href="https://algo-cancer.github.io/team/">Team</a></li>
      	
      
    
  
    
      
      	
      	<li><a href="https://algo-cancer.github.io/papers/">Publications</a></li>
      	
      
    
  
    
      
      	
      	<li><a href="https://algo-cancer.github.io/projects/">Software</a></li>
      	
      
    
  



          <!--<li class="dropdown nav navbar-nav">-->
            <!--<a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">-->
              <!--More <span class="caret"></span>-->
            <!--</a>-->

            <!--<ul class="dropdown-menu nav navbar-nav" role="menu">-->
              <!---->
              <!---->
              <!--


  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
  
    
  
    
  
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  


-->
            <!--</ul>-->
          <!--</li>-->
        </ul>
      </div><!-- /.navbar-collapse -->
    </nav>
   </div>

   <!-- Content -->
   <div class="container">
      
<div class="bigspacer"></div>
<div class="bigspacer"></div>

<div class="row">
	<div class="col-md-12">
		<div class="page-header">
  			<div class="title media-heading">lordFAST</div>
  			
		</div>
	</div>
</div>

<div class="bigspacer"></div>

<div class="row">
	<div class="col-md-3">
		<div class="bigspacer"></div>
		<div class="glyphbox note">
			<div class="pad-left note">
				<div class="smallspacer"></div>
				<i class="fa fa-github-alt fa-fw"></i>
				<a class="off" href="https://github.com/vpc-ccg/lordfast">Source Code</a>
			</div>
			<div class="bigspacer"></div>
		</div>
	</div>
	<div class="col-md-8">
    	<div class="content">
      	<h1 id="lordfast"><strong>lordFAST</strong></h1>

<p>lordFAST is a sensitive tool for mapping long reads with high error rates.
lordFAST is specially designed for aligning reads from PacBio sequencing technology but provides the  user  the ability to change alignment parameters depending on the reads and application.</p>

<h2 id="how-to-install">How to install?</h2>

<p><a href="https://travis-ci.org/vpc-ccg/lordfast"><img src="https://travis-ci.org/vpc-ccg/lordfast.svg?branch=master" alt="Build Status" /></a></p>

<h3 id="requirements">Requirements</h3>
<ul>
  <li>GCC ≥ 4.4.7</li>
  <li>zlib</li>
</ul>

<h3 id="using-conda">Using conda</h3>
<p>lordFast can be installed using <a href="https://conda.io">conda</a> package manager via <a href="https://bioconda.github.io">bioconda</a> channel:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>$ conda install -c bioconda lordfast
</code></pre></div></div>

<h3 id="from-source-code">From source code</h3>
<p>In order to build lordFAST, please download the latest release from <a href="https://github.com/vpc-ccg/lordfast/releases">https://github.com/vpc-ccg/lordfast/releases</a> or alternatively clone the repository by running the following command:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>$ git clone https://github.com/vpc-ccg/lordfast.git Now the code can be compiled easily by running `make` command line which builds the binary file `lordfast`.

$ cd lordfast
$ make
</code></pre></div></div>

<h2 id="how-to-run">How to run?</h2>

<h3 id="synopsis">SYNOPSIS</h3>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>lordfast --index FILE [OPTIONS]
lordfast --search FILE --seq FILE [OPTIONS]
</code></pre></div></div>

<h3 id="options">OPTIONS</h3>
<p>Run <code class="language-plaintext highlighter-rouge">lordfast -h</code> or <code class="language-plaintext highlighter-rouge">man ./HELP.man</code> to see available options.</p>

<h4 id="indexing-options">Indexing options</h4>
<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>-I, --index STR
    Path to the reference genome file in FASTA format which is supposed to be indexed. [required]
</code></pre></div></div>

<h4 id="mapping-options">Mapping options</h4>
<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>-S, --search STR
    Path to the reference genome file in FASTA format. [required]

-s, --seq STR
    Path to the file containing read sequences in FASTA/FASTQ format. [required]

-o, --out STR
    Write output to STR file rather than standard output. [stdout]

-t, --threads INT
    Use INT number of CPU cores. Pass 0 to use all the available cores. [1]
</code></pre></div></div>

<h4 id="advanced-options">Advanced options</h4>
<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>-k, --minAnchorLen INT
    Minimum required length of anchors to be considered. [14]

-n, --numMap INT
    Perform alignment for at most INT candidates. [10]

-l, --minReadLen INT
    Do not try to map any read shorter than INT bp and report them as unmapped. [1000]

-c, --anchorCount INT
    Consider INT anchoring positions on the long read. [1000]

-m, --maxRefHit INT
    Ignore anchoring positions with more than INT reference hits. [1000]

-R, --readGroup STR
    SAM read group line in a format like '@RG\tID:foo\tSM:bar'. []

-a, --chainAlg INT
    Chaining algorithm to use. Options are "dp-n2" and "clasp". [dp-n2]

--noSamHeader
    Do not print sam header in the output.
</code></pre></div></div>

<h4 id="other-options">Other options</h4>
<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>-h, --help
    Prints this help file.

-v, --version
    Prints the version of software.
</code></pre></div></div>

<h3 id="examples">EXAMPLES</h3>
<h4 id="indexing-reference-genome">Indexing reference genome:</h4>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>$ ./lordfast --index refgen.fasta
</code></pre></div></div>

<h4 id="mapping-to-the-reference-genome">Mapping to the reference genome:</h4>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>$ ./lordfast --search refgen.fa --seq reads.fastq &gt; map.sam
$ ./lordfast --search refgen.fa --seq reads.fastq --threads 4 &gt; map.sam
</code></pre></div></div>

<h2 id="publication">Publication</h2>
<p><em>Haghshenas E., Sahinalp S.C. and Hach F.</em>, “lordFAST: sensitive and Fast Alignment Search Tool for LOng noisy Read sequencing Data” Bioinformatics (2018) DOI: <a href="https://doi.org/10.1093/bioinformatics/bty544">10.1093/bioinformatics/bty544</a></p>

<h2 id="bugs">Bugs</h2>
<p>Please report the bugs through lordFAST’s issues page at <a href="https://github.com/vpc-ccg/lordfast/issues">https://github.com/vpc-ccg/lordfast/issues</a>.</p>

<h2 id="contact">Contact</h2>
<p>Ehsan Haghshenas (ehaghshe AT sfu DOT ca)</p>

<h2 id="copyright-and-license">Copyright and License</h2>
<p>This software is released under GNU General Public License (v3.0)<br />
Copyright (c) 2018 Simon Fraser University</p>
<ul>
  <li>BWA (used for the BWT-based index) is developed by Heng Li and is licensed under GPL</li>
  <li>Edlib (used for global alignment) is developed by Martin Sosic and is licensed under MIT</li>
  <li>ksw (used for alignment extension) is licensed under MIT</li>
  <li>clasp (can be used for chaining) is developed and copyrighted by Christian Otto</li>
</ul>

    	</div> 
	</div>
	<div class="col-md-1"></div>
</div>
<div class="bigspacer"></div>


   </div>


    



    <!-- Latest compiled and minified JavaScript, requires jQuery 1.x (2.x not supported in IE8) -->
    <!-- Placed at the end of the document so the pages load faster -->
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.10.2/jquery.min.js"></script>
    <script src="https://algo-cancer.github.io/assets/themes/lab/bootstrap/js/bootstrap.min.js"></script>
    <script type='text/javascript' src='https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js'></script>
  </body>
</html>

