
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">

    <title>Orman | Algorithms for Cancer</title>
    <meta name="description" content="Cenk Sahinalp Lab">
    <meta name="author" content="Cenk Sahinalp">

    <!-- Enable responsive viewport -->
    <meta name="viewport" content="width=device-width, initial-scale=1.0"/>

    <!-- Bootstrap styles -->
    <link href="https://algo-cancer.github.io/assets/themes/lab/bootstrap/css/bootstrap.min.css" rel="stylesheet"/>
    <!-- Optional theme -->
    <link href="https://algo-cancer.github.io/assets/themes/lab/bootstrap/css/bootstrap-theme.min.css" rel="stylesheet"/>
    <!-- Sticky Footer -->
    <link href="https://algo-cancer.github.io/assets/themes/lab/bootstrap/css/bs-sticky-footer.css" rel="stylesheet"/>
    
    <!-- Custom styles -->
    <link href="https://algo-cancer.github.io/assets/themes/lab/css/style.css?body=1" rel="stylesheet" type="text/css" media="all"/>

    <!-- icon -->
    <link rel="icon" type="image/png" href="https://algo-cancer.github.io/assets/themes/lab/images/logo/favicon.png"/>

    <!-- Custom fonts and icons via Font Awesome, http://fortawesome.github.io/Font-Awesome/ -->
    <link rel="stylesheet" href="http://maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css"/>

    <!-- Fonts via Google -->
    <link href='http://fonts.googleapis.com/css?family=Lato:300italic,700italic,300,700' rel='stylesheet' type='text/css'/>

    <!-- Math via MathJax -->
    <script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>


    <!-- HTML5 Shim and Respond.js IE8 support of HTML5 elements and media queries -->
    <!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
    <!--[if lt IE 9]>
      <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
      <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
    <![endif]-->

    <!-- Fav and touch icons -->
    <!-- Update these with your own images
      <link rel="shortcut icon" href="images/favicon.ico">
      <link rel="apple-touch-icon" href="images/apple-touch-icon.png">
      <link rel="apple-touch-icon" sizes="72x72" href="images/apple-touch-icon-72x72.png">
      <link rel="apple-touch-icon" sizes="114x114" href="images/apple-touch-icon-114x114.png">
    -->

    <!-- atom & rss feed
    <link href="/atom.xml" type="application/atom+xml" rel="alternate" title="Sitewide ATOM Feed">
    <link href="/rss.xml" type="application/rss+xml" rel="alternate" title="Sitewide RSS Feed">
    -->

  </head>

  <body>

  <!-- Static top navbar -->

  <nav class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class='container'>
      <!-- Brand and toggle get grouped for better mobile display -->
      <div class="navbar-header">
        <button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#jb-navbar-collapse">
          <span class="sr-only">Toggle navigation</span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
        <a href="/">
        <img class="pull-left logo" src="https://algo-cancer.github.io/images/logo.png">
        </a>
      </div>

      <!-- Collect the nav links, forms, and other content for toggling -->
      <div class="collapse navbar-collapse" id="jb-navbar-collapse">
        <ul class="nav navbar-nav navbar-left hidden-sm">
          <!-- Don't put site title on home page -- there's already a big title there. -->
          
          
          
          <a class="navbar-brand" href="https://algo-cancer.github.io/">Algorithms for Cancer</a>
          
        </ul>
        <ul class="nav navbar-nav navbar-right pull-right">
          
          
          


  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
  
    
  
    
  
    
  
    
      
    
  
    
      
    
  
    
      
      	
      	<li><a href="https://algo-cancer.github.io/research">Research</a></li>
      	
      
    
  
    
      
      	
      	<li><a href="https://algo-cancer.github.io/blog/">Blog</a></li>
      	
      
    
  
    
      
      	
      	<li><a href="https://algo-cancer.github.io/team/">Team</a></li>
      	
      
    
  
    
      
      	
      	<li><a href="https://algo-cancer.github.io/papers/">Publications</a></li>
      	
      
    
  
    
      
      	
      	<li><a href="https://algo-cancer.github.io/projects/">Software</a></li>
      	
      
    
  



          <!--<li class="dropdown nav navbar-nav">-->
            <!--<a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">-->
              <!--More <span class="caret"></span>-->
            <!--</a>-->

            <!--<ul class="dropdown-menu nav navbar-nav" role="menu">-->
              <!---->
              <!---->
              <!--


  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
  
    
  
    
  
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  
    
      
    
  


-->
            <!--</ul>-->
          <!--</li>-->
        </ul>
      </div><!-- /.navbar-collapse -->
    </nav>
   </div>

   <!-- Content -->
   <div class="container">
      
<div class="bigspacer"></div>
<div class="bigspacer"></div>

<div class="row">
	<div class="col-md-12">
		<div class="page-header">
  			<div class="title media-heading">Orman</div>
  			
		</div>
	</div>
</div>

<div class="bigspacer"></div>

<div class="row">
	<div class="col-md-3">
		<div class="bigspacer"></div>
		<div class="glyphbox note">
			<div class="pad-left note">
				<div class="smallspacer"></div>
				<i class="fa fa-github-alt fa-fw"></i>
				<a class="off" href="https://github.com/sfu-compbio/orman">Source Code</a>
			</div>
			<div class="bigspacer"></div>
		</div>
	</div>
	<div class="col-md-8">
    	<div class="content">
      	<h1 id="orman">ORMAN</h1>
<p>a.k.a. <i><b>O</b>ptimal <b>R</b>esolution of <b>M</b>ultimapping <b>A</b>mbiguity of R<b>N</b>A-Seq Reads</i></p>
<hr />

<h1 id="introduction">Introduction</h1>

<h3 id="eh-orman">Eh… ORMAN?</h3>

<p>ORMAN is a tool for resolving multi-mappings within an RNA-Seq SAM file. ORMAN was publised in <a href="http://bioinformatics.oxfordjournals.org/content/30/5/644">Bioinformatics in October 2013</a>.</p>

<h3 id="how-do-i-get-orman">How do I get ORMAN?</h3>

<p>Just clone our repository and issue <code class="language-plaintext highlighter-rouge">make</code> command:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>git clone https://github.com/sfu-compbio/orman.git
cd orman &amp;&amp; make -j
</code></pre></div></div>

<p>This will work only if you have your CPLEX installed in <code class="language-plaintext highlighter-rouge">/opt/ibm/cplex12_5_1</code>. If not, keep reading, or just try out our x86-64 <a href="https://github.com/sfu-compbio/orman/tarball/binary">binary</a> (compiled on CentOS 7.x).</p>

<h3 id="how-do-i-get-cplex">How do I get CPLEX?</h3>

<p>You will need to install IBM CPLEX Optimization Studio (preferably versions &gt;= 12.5.1) before compiling ORMAN. You can get the trial version at <a href="http://www-01.ibm.com/software/websphere/products/optimization/cplex-studio-preview-edition/">IBM’s website</a>.</p>

<p>If you are student or a faculty member, you can get CPLEX for free (woohoo!). Just sign up for <a href="http://www-03.ibm.com/ibm/university/academic/pub/page/academic_initiative">Academic Initiative Program</a>.</p>

<p>After you are done with installing your CPLEX, please set the <code class="language-plaintext highlighter-rouge">CPLEXDIR</code> variable to point to the CPLEX installation directory. For example, if you have installed CPLEX in <code class="language-plaintext highlighter-rouge">~/apps/mycplex</code>, run <code class="language-plaintext highlighter-rouge">make</code> as follows:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>CPLEXDIR=~/apps/mycplex make
</code></pre></div></div>

<p>If you are not using x86-64 system, you will also need to set up <code class="language-plaintext highlighter-rouge">CPLEXARCH</code> variable. Please refer to the CPLEX documentation to discover its proper value (but please note that we cannot guarantee that ORMAN will run properly on such systems).</p>

<h3 id="how-do-i-run-orman">How do I run ORMAN?</h3>

<p>ORMAN is invoked as following:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>orman -t [thread count] -g [ORMAN gtf file] -s [sam file] [output file]
</code></pre></div></div>

<p>Example invocation (results will be stored in <code class="language-plaintext highlighter-rouge">orman_awesome.sam</code> file):</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>orman -t 10 -g super.gtf -s awesome.sam orman_awesome.sam
</code></pre></div></div>

<p><strong>Please note that input SAM file should be sorted by the read name, and it should have valid headers.</strong></p>

<h4 id="ormangtf">ormanGTF</h4>

<p>ORMAN supports only GTF files containing ORMAN-specific additional information (namely <code class="language-plaintext highlighter-rouge">partial_ex</code> field). If you don’t have such file, use the provided <code class="language-plaintext highlighter-rouge">ormanGTF</code> script to create compatinge GTF file as follows:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>ormanGTF &lt; [input gtf file] &gt; [new gtf file]
</code></pre></div></div>

<p>Example:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>ormanGTF &lt; almost_super.gtf &gt; super.gtf
</code></pre></div></div>

<h1 id="usage">Usage</h1>

<h3 id="parameter-explanation">Parameter explanation</h3>

<ul>
  <li>
    <p><code class="language-plaintext highlighter-rouge">--help, -h</code></p>

    <p>Display not-so-cute help message and exit gracefully.</p>
  </li>
  <li>
    <p><code class="language-plaintext highlighter-rouge">--threads, -t [number]</code></p>

    <p>Set up the maximum number of threads ORMAN may use.  ORMAN uses multi-threading to speed up CPLEX smoothing step.</p>

    <p>Beware that CPLEX may require a significant amount of memory to complete, and that large number of threads may result in large memory consumption.</p>

    <p>Default value: <strong>number of available processors</strong></p>
  </li>
  <li>
    <p><code class="language-plaintext highlighter-rouge">--sam, -s [file]</code></p>

    <p>Specify input SAM file here.
Requirements for SAM file are:</p>

    <ul>
      <li>Header must be present (otherwise ORMAN may behave strange …)</li>
      <li>All reads with the same read name have to be clustered together.</li>
    </ul>

    <p>Or simply: SAM file should be sorted by the read name.
If not, ORMAN WILL NOT report any error, but the results might (and will) be interesting.</p>
  </li>
  <li>
    <p><code class="language-plaintext highlighter-rouge">--gtf, -g [file]</code></p>

    <p>Specify input GTF file here.
ORMAN supports only GTF files containing ORMAN-specific additional information (namely <code class="language-plaintext highlighter-rouge">partial_ex</code> field). If you don’t have such file, please use the provided ormanGTF script to create one.</p>
  </li>
  <li>
    <p><code class="language-plaintext highlighter-rouge">--single-end, -1</code></p>

    <p>Let ORMAN treat input SAM file as a single-end mapping (i.e. assume that no paired-end mapping was performed).</p>
  </li>
</ul>

<h3 id="some-additional-information">Some additional information</h3>

<p>ORMAN adds <code class="language-plaintext highlighter-rouge">YG</code> tag to each read. <code class="language-plaintext highlighter-rouge">YG</code> denotes the gene to which the read was assigned by ORMAN. If ORMAN couldn’t handle read (some weird fusion involved, or it is just crappy read), <code class="language-plaintext highlighter-rouge">YG</code> tag will be assigned to <code class="language-plaintext highlighter-rouge">_</code>.</p>

<p>Every resolved read will have its <code class="language-plaintext highlighter-rouge">NH</code> tag set to 1. Also, for every resolved read, <code class="language-plaintext highlighter-rouge">CC</code>, <code class="language-plaintext highlighter-rouge">CP</code> and <code class="language-plaintext highlighter-rouge">HI</code> tags will be removed.</p>

<p>If ORMAN complains about too many crappy reads, please make sure that GTF and SAM chromosome names march. Common cause is the mismatch caused by <code class="language-plaintext highlighter-rouge">chr</code> prefix (e.g. <code class="language-plaintext highlighter-rouge">1</code> vs <code class="language-plaintext highlighter-rouge">chr1</code>).</p>

<blockquote>
  <p><strong>A friendly advice</strong>: ORMAN might use easily &gt;15G memory if the input is quite large SAM file.</p>
</blockquote>

<h1 id="support">Support</h1>

<h3 id="contact--support">Contact &amp; Support</h3>

<p>Feel free to drop any inquiry to <a href="mailto:inumanag@">inumanag at sfu dot oh canada</a>.</p>

<h3 id="authors">Authors</h3>

<p>ORMAN has been brought to you by:</p>

<ul>
  <li><a href="http://www.cs.sfu.ca/~pdao/personal">Dr. Phuong Dao</a></li>
  <li><a href="http://www.sfu.ca/~inumanag">Ibrahim Numanagić</a></li>
  <li><a href="http://www.sfu.ca/~yenyil/">Yen-Yi Lin</a></li>
  <li><a href="http://www.cs.sfu.ca/~fhach/personal/">Dr. Faraz Hach</a></li>
  <li><a href="http://www.cs.toronto.edu/~nild">Dr. Nilgun Donmez</a></li>
</ul>

<p>from the <a href="http://compbio.cs.sfu.ca">Lab for Computational Biology</a> at <a href="http://www.sfu.ca">Simon Fraser University</a> and <a href="http://eichlerlab.gs.washington.edu/">Eicher Lab</a> at <a href="http://www.washington.edu">University of Washington</a>.</p>

<h3 id="licence">Licence</h3>

<p>Copyright (c) 2012–2013, Simon Fraser University. All rights reserved.</p>

<p>Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:</p>

<ul>
  <li>Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.</li>
  <li>Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.</li>
  <li>Neither the name of the Simon Fraser University nor the names of its contributors may be used to endorse or promote products derived from this software without specific prior written permission.</li>
</ul>

<p>THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS “AS IS” AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT	LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR	A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR	CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,	EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,	PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR	PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF	LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING	NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS	SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.</p>

<h1 id="release-notes">Release notes</h1>

<ul>
  <li><em>(15-Nov-2015)</em> <strong>ORMAN version 1.3 release fix</strong>
    <ul>
      <li>Added single-end mode (<code class="language-plaintext highlighter-rouge">-1</code>) for samples which do not contain paired-end reads.</li>
    </ul>
  </li>
  <li><em>(12-Jun-2014)</em> <strong>ORMAN version 1.2 release fix</strong>
    <ul>
      <li>Additional headers are added so that ORMAN can be compiled on systems without boost libraries.</li>
      <li>ormanGTF script is now working on unix-systems without any modifications.</li>
    </ul>
  </li>
  <li><em>(06-Mar-2014)</em> <strong>ORMAN version 1.2 release</strong>
    <ul>
      <li>NH tags are properly updated.</li>
      <li>CC, CP and HI tags are removed if the read is resolved.</li>
    </ul>
  </li>
  <li><em>(14-Jan-2014)</em> <strong>ORMAN version 1.1.1 bugfix release</strong>
    <ul>
      <li>A serious bug was found in the ILP formulation function, which was fixed in this release. Please upgrade to v1.1.1 as soon as possible.</li>
    </ul>
  </li>
  <li><em>(24-Oct-2013)</em> <strong>Compile error fixes</strong>
    <ul>
      <li>The previously posted source code archive was missing few header files necessary for the compilation process. The missing files are added back now. Please re-download the source code if this bug affected you.</li>
    </ul>
  </li>
  <li><em>(30-Sep-2013)</em> <strong>ORMAN version 1.1 release</strong>
    <ul>
      <li>Initial public release</li>
    </ul>
  </li>
</ul>


    	</div> 
	</div>
	<div class="col-md-1"></div>
</div>
<div class="bigspacer"></div>


   </div>


    



    <!-- Latest compiled and minified JavaScript, requires jQuery 1.x (2.x not supported in IE8) -->
    <!-- Placed at the end of the document so the pages load faster -->
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.10.2/jquery.min.js"></script>
    <script src="https://algo-cancer.github.io/assets/themes/lab/bootstrap/js/bootstrap.min.js"></script>
    <script type='text/javascript' src='https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js'></script>
  </body>
</html>

